dev/com.ibm.ws.microprofile.openapi_fat/fat/src/com/ibm/ws/microprofile/openapi/fat/config/OpenAPIConfigTest.java

```diff
@@ -255,6 +258,53 @@ public void testApplicationPathConfict() throws Exception {
         assertMissing("/mpOpenAPIConfigTest");
     }
 
+    @Test
+    @Mode(TestMode.FULL)
+    public void testCustomizedEndpointProxyRefererHeader() throws Exception{
+        // Defaults are tested in the ProxySupportTest.class
+        ServerConfiguration config = server.getServerConfiguration();
+        config.getMpOpenAPIElement().setDocPath("/foo");
+        config.getMpOpenAPIElement().setUiPath("/bar");
+        server.updateServerConfiguration(config);
+
+        server.startServer(false);
+
+        assertWebAppStarts("/foo");
+        assertWebAppStarts("/bar");
+
+        assertDocumentPath("/foo");
+        assertUiPath("/bar");
+
+        // Test changed path with various referer headers
+
+        //Default HTTP port with matching docPath
+        OpenAPI model = new OpenAPIConnection(server, "/foo").header("Referer","http://testurl1/foo").downloadModel();
+        //as ports is not a liberty report, only return a single server
+        assertTrue("Check that the list of servers is not empty",model.getServers().size()==1);
+        //check that the server name has changed to allow for either http:// or https:// to be first
+        assertTrue("check that the servers entry use the server from the referer header",model.getServers().get(0).getUrl().contains("http://testurl1/"+APP_NAME));
```

Comment:
Use `assertThat` rather than `assertTrue` so that you get better error messages.